File: internal/models/asset.go
package models

import "time"

type Asset struct {
	ID   int
	Name string `json:"name"`
	Type string `json:"type"`

	Ticker               string    `json:"ticker,omitempty"`
	Price                float32   `json:"price,omitempty"`
	Amount               float32   `json:"amount,omitempty"`
	Currency             string    `json:"currency,omitempty"`
	InterestStart        time.Time `json:"interestStart,omitempty"`
	InterestRate         float32   `json:"interestRate,omitempty"`
	CompoundingFrequency string    `json:"compoundingFrequency,omitempty"`
}

File: pfinance-web/src/types/asset.ts
export interface Asset {
    id?: number;
    type: "Stock" | "Gold" | "Bond" | "Savings" | "Crypto";
    name: string;
    ticker?: string;
    price?: number;
    amount: number;
    currency: string;
    interestRate?: number;
    compoundingFrequency?: "daily" | "monthly" | "quarterly" | "annually";
    interestStart?: string;
    createdAt: string;
  }
